.github/workflows/install_test.yaml

```diff
@@ -37,16 +34,16 @@ jobs:
         with:
           path: ${{ env.pythonLocation }} 
           key: ${{ matrix.os- }}-${{ matrix.python_version }}-install-${{ env.pythonLocation }}-${{ hashFiles('**/pyproject.toml') }}-v01
+      - name: Set up Homebrew
+        id: set-up-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+      - name: Set up cmake and libomp
+        run: brew install cmake libomp
       - name: Build evalml package
         run: make package
       - name: Install evalml from sdist (not using cache)
-        if: steps.cache.outputs.cache-hit != 'true'
         run: |
           python -m pip install "unpacked_sdist/."
-      - name: Install evalml from sdist (using cache)
-        if: steps.cache.outputs.cache-hit == 'true'
-        run: |
-          python -m pip install "unpacked_sdist/." --no-deps
```

Comment:
Whenever the tests would use cache, the test would fail.

Reply:
Interesting, do you think it's noticably slower since we're not using the cache? If so, by how much?

Reply:
Interestingly the time it takes varies.

The latest commit ([2b11a69](https://github.com/alteryx/evalml/pull/4423/commits/2b11a6918d3620df5eee8426390626714cb3bd47)) has the time that ubuntu takes for this step to be 8 seconds. For MacOS it takes 40 seconds for python3.9 and 3 minutes for python3.11.

Conversely, the second latest commit ([cd3f4f8](https://github.com/alteryx/evalml/pull/4423/commits/cd3f4f8110eb2d1053ab4443f993a947c344e9a3)) has the time that ubuntu takes hover around 1 minute. MacOS hovers at around 40 seconds.

I can't say for sure how much slower it is without cache because it didn't work with cache enabled. I can go test to see if I can get cache working (hopefully it doesn't erase the approvals).

Reply:
FWIW, I was having trouble with caches in other OS repos recently and ended up removing them because they were failing too often.